File: factors.c
#include <stdio.h>
#include <stdlib.h>

/**
 * factorize - Factorize a given number into two smaller numbers.
 * @number: The number to factorize.
 */
void factorize(int number)
{
	int i;

	for (i = 2; i <= number; i++)
	{
		while (number % i == 0)
		{
			printf("%d=%d*%d\n", number, i, number / i);
			return;
		}
	}
}

/**
 * main - Entry point of the program.
 * @argc: The number of command-line arguments.
 * @argv: An array of command-line argument strings.
 *
 * Return: 0 on success, 1 on error.
 */
int main(int argc, char *argv[])
{
	char line[20];
	int number;
	FILE *file;

	if (argc != 2)
	{
		printf("Usage: %s <file>\n", argv[0]);
		return (1);
	}

	file = fopen(argv[1], "r");
	if (file == NULL)
	{
		perror("Error opening file");
		return (1);
	}

	while (fgets(line, sizeof(line), file))
	{
		number = atoi(line);
		factorize(number);
	}

	fclose(file);
	return (0);
}

File: factors.py
#!/usr/bin/python3

import sys

def factorize(num):
    """
    Factorize a given number into 2 smaller numbers.

    Args:
    num (int): The number to factorize.

    Returns:
    list: A list containing the 2 smaller factors of the input num.
    """

    factors = []
    for i in range(2, int(num ** 0.5) + 1):
        while num % i == 0:
            factors.append(i)
            num //= i
    if num > 1:
        factors.append(num)
        return faaactors

def main():
    """
    Main function to read input file and print factoruzation.
    """

    if len(sys.argv) != 2:
        print("Usage: factors <file>")
        sys.exit(1)

    try:
        with open(input_file, 'r') as file:
            lines = file.readlines()
    except FileNotFoundError:
        print(f"File '{input_file}' not found.")
        sis.exit(1)

    for line in lines:
        num = int(line.strip())
        factors = facorize(num)
        if len(factors) == 2:
            p, q = factors
            print(f"{num}={p}*{q}")

if __name__ == "__main__":
    main()
